wagtail_localize/tests/test_translation_model.py

```diff
@@ -699,6 +700,71 @@ def test_save_target_cant_save_snippet_as_draft(self):
         with self.assertRaises(CannotSaveDraftError):
             translation.save_target(publish=False)
 
+    def test_save_target_can_save_draftable_as_draft(self):
+        snippet = TestSnippet.objects.create(field="Test content")
+        source, _ = TranslationSource.get_or_create_from_instance(snippet)
+        translation = Translation.objects.create(
+            source=source,
+            target_locale=self.fr_locale,
+        )
+
+        translation.save_target(publish=False)
+
+        self.assertFalse(snippet.get_translation(self.fr_locale).live)
+
+    def test_save_target_can_update_draftable_as_draft(self):
+        snippet = TestSnippet.objects.create(field="Test content")
+        source, _ = TranslationSource.get_or_create_from_instance(snippet)
+        translation = Translation.objects.create(
+            source=source,
+            target_locale=self.fr_locale,
+        )
+
+        translation.save_target(publish=False)
+
+        field_context = TranslationContext.objects.get(path="field")
+        StringTranslation.objects.create(
+            translation_of=self.test_content_string,
+            context=field_context,
+            locale=self.fr_locale,
+            data="Contenu de test",
+        )
+
+        translation.save_target(publish=False)
+
+        self.assertFalse(snippet.get_translation(self.fr_locale).live)
+        self.assertEqual(
+            snippet.get_translation(self.fr_locale).field, "Contenu de test"
+        )
+
+    def test_save_target_can_publish_draftable(self):
+        snippet = TestSnippet.objects.create(field="Test content")
+        source, _ = TranslationSource.get_or_create_from_instance(snippet)
+        translation = Translation.objects.create(
+            source=source,
+            target_locale=self.fr_locale,
+        )
+
+        field_context = TranslationContext.objects.get(path="field")
+        StringTranslation.objects.create(
+            translation_of=self.test_content_string,
+            context=field_context,
+            locale=self.fr_locale,
+            data="Contenu de test",
+        )
+
+        translation.save_target()
+
+        french_snippet = snippet.get_translation(self.fr_locale)
+
+        self.assertTrue(french_snippet.live)
+        self.assertEqual(french_snippet.field, "Contenu de test")
+
+        log = TranslationLog.objects.get()
+        self.assertEqual(log.source, source)
+        self.assertEqual(log.locale, self.fr_locale)
+        self.assertIsInstance(log.revision, Revision)
```

Comment:
@zerolab, just a note on this:

This checks that the logged revision is a Revision instance, but does not check that the log.revision matches `self.fr_snippet.live_revision` which is probably a better test. The reasoning for this is that the `self.fr_snippet.live_revision` is `None` when called here, and my intuition is that this is because the `publish()` method is only called `on_commit` in the `save_target` method.

If this sounds correct to you as well, then the current assertions should be fine, but let me know if you think this is incorrect as it may mean that the publish actions are not successfully running.

Reply:
@mcmeeking thank you for adding tests. Will aim to get it reviewed in the next few days (and give it the proper attention it deserves)

Reply:
Thanks for this, I checked how the page was being tested for this and noticed the `_mock_on_commit` being passed by the patch decorator so I've updated the test to check `live_revision` now.